### PR TITLE
Typed array support for XMLHttpRequest's send API

### LINKED ISSUE #20343: Implement typed array support for XMLHttpRequest's send API
Now that #20267 has merged, we can uncomment the [BufferSource](https://github.com/servo/servo/blob/master/components/script/dom/webidls/XMLHttpRequest.webidl#L16) argument to allow XMLHttpRequest.send() to support sending typed arrays as the XHR body. This will require extracting the data from the typed array in [this code](https://github.com/servo/servo/blob/bedeec0cf54011e3d15bd50ba32b6b422958b483/components/script/dom/xmlhttprequest.rs#L507-L523).

Code: `components/script/dom/webidls/XMLHttpRequest.webidl`, `components/script/dom/xmlhttprequest.rs`
Tests: `./mach test-wpt tests/wpt/web-platform-tests/xhr/` (this should generate a bunch of new [test pass results](https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations))

### PR DESCRIPTION
**Pull request by nupurbaghel**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #20343 

- [x] Updated some tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20434)

### CODE REVIEW

**Comment by jdm** [comment]
@bors-servo r+
Thanks, this looks like a good improvement!

**Comment by jdm** [comment]
Looks like we now pass /fetch/api/request/request-consume-empty.html as well :)

**Comment by nupurbaghel** [comment]
The test for /fetch/api/request/request-consume.html passed as well. But I also saw some css files failing. Should I leave them as it is?

**Comment by jdm** [comment]
I'm not sure what you mean by css files failing. You should definitely update the ini files for request-consume.html and request-consume-empty.html though.

**Comment by jdm** [comment]
So many more passing tests \o/

You can also go ahead and `rm  tests/wpt/mozilla/tests/mozilla/send-blob.htm tests/wpt/mozilla/tests/mozilla/send-arraybuffer.htm  tests/wpt/mozilla/tests/mozilla/resources/content.py` and run `./mach update-manifest`. Those are tests that were added in #9971 to address our incomplete implementation; now that the implementation is complete, we don't need them any more.

**Comment by nupurbaghel** [comment]
I think I have updated all the test files now  :) and removed any conflicting files. I hope it passes all tests this time 😅

File: tests/wpt/metadata/fetch/api/request/request-consume.html.ini [hunk 1]
```diff
@@ -6,37 +6,7 @@
...
-    expected: FAIL
-
-  [Consume DataView request's body as blob]
-    expected: FAIL
-
-  [Consume DataView request's body as arrayBuffer]
-    expected: FAIL
-
-  [Consume DataView request's body as JSON]
+  [Trying to consume bad JSON text as JSON: 'undefined']
```
Comment (nupurbaghel) [inline tests/wpt/metadata/fetch/api/request/request-consume.html.ini]: I had a little doubt, whether it should be here ?

Comment (jdm) [reply tests/wpt/metadata/fetch/api/request/request-consume.html.ini]: The easiest way to find out is to run `./mach test-wpt --no-pause-after-test tests/wpt/fetch/api/request/request-consume.html` and see if any unexpected results are reported.

**Comment by nupurbaghel** [comment]
Some tests which were passing earlier are now failing locally on my device. This seems pretty strange as I have rebased against all the recent changes. If the merge fails again, I will try doing manual edits. Hope that is fine 😅 
Thank you.

**Comment by jdm** [comment]
@bors-servo retry
* #20471 #20470
In the interest of getting this landed, we can look into the underlying cause of the intermittent behaviour separately.

**Comment by nupurbaghel** [comment]
Now since we have separate issues to handle this, should I add the updated PASS in request-consume and request-consume-empty files here?

**Pull request merged**